rust/sbp2json/src/lib.rs

```diff
@@ -78,7 +79,13 @@ where
     W: Write,
     F: Formatter + Clone,
 {
-    let source = maybe_fatal_errors(sbp::iter_messages(input), fatal_errors);
+    let source = maybe_fatal_errors(sbp::iter_frames(input), fatal_errors).map(|frame| {
+        frame.to_sbp().unwrap_or(Sbp::Unknown(Unknown {
```

Comment:
I would probably implement Serialize/Deserialize for Frame instead of using Unknown for the case when the frame is invalid so we can distinguish between an Unknown message type and in invalid message. It should look the same (as in serialize/deserialize to a struct with msg_type, sender_id and payload fields) but then you can tell whether or not its an issue with an outdated version of libsbp or a malformed/corrupted message.

Also you'll need to handle the other direction for this is json2sbp

Reply:
although I guess we are using an untagged enum so you wouldn't be able to tell the difference between a Frame and an Unknown hmmm

Reply:
hmmmm, is unknown actually used for anything tho
could just use unknown for the types of msg that we dont known about

this would make it so we dont need to do anything to json2sbp since the msgs they produce would just be kinda the payload except wrapped with unknown 🤔 

unless we want exact raw corrupt packet without wrapping anything then might be a bit more complicated than this.

Reply:
@notoriaga i.e. would it be ok to simply wrap around as an unknown message (since these msgs are indeed 'unknown') or do we want some special form of deserialization in JSON for corrupt messages in addition to the sbp?
i.e.
```json
{
    msg_type: ...
    payload: ...
    sender_id: ....
}
{
    corrupt_data: ...
}
```
or simply just handling corrupt_data inside Unknown packets like
```json
{
    msg_type: 0
    sender_id: 0
    payload: the corrupt msg
}
```

@jbangelo @silverjam

Reply:
We could also add a msg_name field to the json representation of messages so then we'd have 

```
{
    "msg_type": 123,
    "msg_name": Unknown,
    "sender_id": 4,
    "payload": [...]
}
{
    "msg_type": 321,
    "msg_name": Frame,
    "sender_id": 4,
    "payload": [...]
}
```

could also be nice to have for jq stuff, the message names are bit easier to remember than the ids

Reply:
think the message names are alread yhtere for sbp2json, making this field necessary would mean json2sbp only uses this field to check specific case where `msg_name == "FRAME"` 🤔 not sure if theres any better way or...
^^ this also seems much more complicated / invovled tho #1317 WIP
hmmmm....